lib/datadog/tracing/contrib/active_job/log_injection.rb

```diff
@@ -6,17 +6,31 @@ module Contrib
       module ActiveJob
         # Active Job log injection wrapped around job execution
         module LogInjection
-          def self.included(base)
-            base.class_eval do
-              around_perform do |_, block|
-                if Datadog.configuration.tracing.log_injection && logger.respond_to?(:tagged)
-                  logger.tagged(Tracing.log_correlation, &block)
-                else
-                  block.call
+          # Active Job 4 / 5 don't execute `perform_now` at the right point, so we do best effort log correlation tagging
+          module ActiveJob4
```

Comment:
One minor suggestion:

What do you think, does it make sense to avoid the duplicated "ActiveJob" in the class name `ActiveJob::LogInjection::ActionJobX` and name the module by what it does? I'm thinking of something like `ActiveJob::LogInjection::AroundPerformPatch` and `ActiveJob::LogInjection::TaggedPatch`?